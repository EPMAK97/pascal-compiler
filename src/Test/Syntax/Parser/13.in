var x : integer;
Begin
    x := 1;
    if x < 10 then begin
        x := x * 2 + 4;
    end
end.