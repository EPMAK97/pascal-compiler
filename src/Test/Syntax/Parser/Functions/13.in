program test;

function f(a, b, c: Double) : Double;
begin
    Result := (a + b) / c;
end;

var

    r: Double;

begin
    r := f(1.0, r, 3);
end.