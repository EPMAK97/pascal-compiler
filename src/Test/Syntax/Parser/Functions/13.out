FUNCTION : test
    result : NIL
    function vars {
    FUNCTION : f
        result : DOUBLE
        function params {
        a : DOUBLE
        b : DOUBLE
        c : DOUBLE
        }
        function vars {
        }
        └── statements
                     └── :=
                          ├── Result
                          └── /
                              ├── +
                              |   ├── a
                              |   └── b
                              └── c

    r : DOUBLE
    }
    └── statements
                 └── :=
                      ├── r
                      └── f
                          └── params
                                   ├── 1.0
                                   ├── r
                                   └── 3

